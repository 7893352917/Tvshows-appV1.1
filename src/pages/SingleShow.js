import React, { useEffect, useState } from 'react';
import { getSingleShow } from '../service/TvshowsApi';

const SingleShow = ({ match }) => {
    const [show, setShow] = useState([])
    useEffect(() => {
        getSingleShow(match.params.id).then((res) => {
            setShow(res.data);
        })
    }, [])

    return (
        <div className="singlepage">
            <img src={show.image ? show.image.medium : "No image"} alt={show.name}/>
            <div className="singlepage__info">
                <h1>{show !== null && show.name}</h1>
                <h3>Rating: {show.rating?.average ? show.rating.average : "No rating"}</h3>
                <h3>{show.genres && show.genres.map(genre => {
                    return <span key={genre} className="singlepage__genre">{genre}</span>
                })}</h3>
                <p>Language: {show.language}</p>
                <p>Status: {show.status && show.status}</p>
                <p>Premiered: {show.premiered ? show.premiered : "N/A"}</p>
                <p>Country: {show.network && show.network.country ? show.network.country.name : "N/A"}</p>
                <p>Official Site:  {show.officialSite ? show.officialSite : "N/A"}</p>
                <p>{show.summary && show.summary.replace(/(<([^>]+)>)/gi, "")}</p>
            </div>
        </div>
    )
}

export default SingleShow;