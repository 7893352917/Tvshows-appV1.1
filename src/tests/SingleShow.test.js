import React from 'react';
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from 'enzyme';
import SingleShow from '../pages/SingleShow';
import { render } from '@testing-library/react';
configure({ adapter: new Adapter() });
const fakeUser = {
  data: {
    "name": "Under the Dome",
    "id": 1,
    "genres": ["Drama", "Science-Fiction", "Thriller"],
    "rating": { "average": 6.6 },
    "image": { "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg" },
    "premiered": "2013-06-24",
    "status": "Ended",
    "officialSite": "http://www.cbs.com/shows/under-the-dome/",
    "network": { "id": 2, "name": "CBS", "country": { "name": "United States", "code": "US", "timezone": "America/New_York" } }
  }
};

jest.mock('../service/TvshowsApi', () => ({
  getSingleShow: jest.fn((a) => Promise.resolve(fakeUser)),
}));

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: (show) => [show, jest.fn()],
}));

describe("SingleShow Component", () => {
  const match = {
    params: { id: 2 }
  }
  let component = shallow((<SingleShow />));
  render(<SingleShow match={match} />);

  it('should have an div field', () => {
    expect(component.find('div').length).toEqual(2);
  });

  it('should have a p tag', () => {
    expect(component.find('p').length).toEqual(6);
  });

  it('should match the snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});