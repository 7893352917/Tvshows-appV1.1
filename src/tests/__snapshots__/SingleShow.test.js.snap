// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleShow Component should match the snapshot 1`] = `"<div class=\\"singlepage\\"><img src=\\"No image\\"/><div class=\\"singlepage__info\\"><h1></h1><h3>Rating: No rating</h3><h3></h3><p>Language: </p><p>Status: </p><p>Premiered: N/A</p><p>Country: N/A</p><p>Official Site:  N/A</p><p></p></div></div>"`;
